source/documentation/how-to/hosted-engine.html.md

```diff
@@ -144,15 +144,40 @@ Assuming you have already deployed Hosted Engine on your hosts and running the H
 
 1.  Set hosted engine maintenance mode to global (now ha agent stop monitoring engine-vm, you can see above how to activate it)
 2.  Access to engine-vm and upgrade oVirt to latest version using the same procedure used for non hosted engine setups.
-3.  Select one of the hosted-engine nodes (hypervisors) and put it into maintenance mode from the engine. Note that the host must be in maintenance to allow upgrade to run.
-4.  Upgrade that host with new packages (changes repository to latest version and run yum update -y) on this stage may appear vdsm-tool exception <https://bugzilla.redhat.com/show_bug.cgi?id=1088805>
-5.  Restart vdsmd (# service vdsmd restart)
-6.  Restart ha-agent and broker services (# systemctl restart ovirt-ha-broker && systemctl restart ovirt-ha-agent)
-7.  Exit the global maintenance mode: in a few minutes the engine VM should migrate to the fresh upgraded host cause it will get an higher score
-8.  When the migration has been completed re-enter into global maintenance mode
-9.  Repeat step 3-6 for all the other hosted-engine hosts
-10. Enter for example via UI to engine and change 'Default' cluster (where all your hosted hosts seats) compatibility version to current version (for example 3.6 and activate your hosts (to get features of the new version)
-11. Change hosted-engine maintenance to none, starting from 3.4 you can do it via UI(right click on engine vm, and 'Disable Global HA Maintenance Mode')
+3.  Host: Enable global maintenance mode from one of the hosts
+```
+host ~ # hosted-engine --set-maintenance --mode=global
+```
+4.  Host: Update packages (you may need to changes repository to latest version or install the latest versions metapackage e.g. ovirt-release42-4.2.1)
+```
+host ~ # yum update
+```
+5.  Host: Restart vdsmd
+```
+host ~ # systemctl restart vdsmd
+```
+6.  Host: Restart ha-agent and broker services
+```
+host ~ # systemctl restart ovirt-ha-broker && systemctl restart ovirt-ha-agent
+```
+7.  Hosted-Engine-VM: Update packages
```

Comment:
You seem to detail here updating the engine on the hosted-engine VM. This is already mentioned in step 2. If you decide to add them here, please do it in/after step 2, not here.

Details are provided elsewhere, e.g. in https://www.ovirt.org/documentation/upgrade-guide/upgrade-guide/ . Basically, these are:

yum update ovirt-engine\*setup\*
engine-setup
yum update

If upgrading to the next minor:
Make sure you are on the latest .z, or upgrade to it using above procedure
Add new .y repos
Follow above procedure

Usually the first can be shortened to 'yum update', but it sometimes fails (and we sometimes fix, but still).

Reply:
> Details are provided elsewhere, e.g. in https://www.ovirt.org/documentation/upgrade-guide/upgrade-guide/

My thinking is - why even have a duplication of these steps then?

- It's two lots of technical documentation to maintain.
- I found this how-to when searching for the process _before_ I found that link.

Should we instead remove this how-to in promotion of the link you've provided?

Reply:
Well, that's an option. The "hosted-engine howto" (current doc) was written before the upgrade guide, and still maintained over the years. It's considered to be a "first place to check" for links and tips for hosted-engine stuff. The upgrade guide is (sadly) not very well maintained. I agree it might make sense to remove the upgrade section from the howto and replace it with a link, but perhaps also verify that the upgrade guide is still up-to-date (and at least update the version numbers there).